src/fsharp/service/service.fs

```diff
@@ -925,9 +927,19 @@ type BackgroundCompiler(
             return options, (diags @ errors.Diagnostics)
           }
           |> Cancellable.toAsync
-            
+
+    member _.GetCachedProjectOptions(projectOrScriptFileName, isScript) =
+        let projectFileName =
+            if isScript then getScriptProjectFileName projectOrScriptFileName
+            else projectOrScriptFileName
+        incrementalBuildersCache.Keys(AnyCallerThread) |> List.filter (fun x -> x.ProjectFileName = projectFileName)
+
     member bc.InvalidateConfiguration(options: FSharpProjectOptions, userOpName) =
         if incrementalBuildersCache.ContainsSimilarKey (AnyCallerThread, options) then
+            parseCacheLock.AcquireLock(fun ltok -> 
+                for sourceFile in options.SourceFiles do
+                    checkFileInProjectCache.RemoveAnySimilar(ltok, (sourceFile, 0L, options))
```

Comment:
Hm, it seems this will also erase cached results in "similar" projects, e.g. there could be cached results for a file in another target framework in the same project. This API shouldn't touch other "similar" projects and only remove cache for the "same" project options, though this shouldn't likely be a real problem in practice. @DedSec256 could you take a look, please?

Reply:
`getOrCreateBuilder` function does similar things and uses `RemoveAnySimilar` too https://github.com/dotnet/fsharp/blob/main/src/fsharp/service/service.fs#L419

Reply:
That's interesting. We could probably change it there too then, but it also shows it's probably not that bad to do it like that this time, since this approach is already used in other places and we haven't noticed it before.